src/hb-serialize.hh

```diff
@@ -279,15 +289,34 @@ struct hb_serialize_context_t
     assert (current);
     assert (current->head <= (const char *) &ofs);
 
+    const char  *dflt_base;
+    auto& link = *current->links.push ();
+
+    switch (mode)
+    {
+    case Head:
+      dflt_base = current->head;
+      link.is_absolute = false;
+      break;
+    case Tail:
+      dflt_base = current->tail;
```

Comment:
This is the problematic part.

Reply:
>whereas the object might be enlarged after. We should do the end-math in resolve_links.

I can do that. In that case, I believe mode should be stored in `link_t` replacing `is_absolute`.

Reply:
> I can do that. In that case, I believe mode should be stored in `link_t` replacing `is_absolute`.

Correct.

We should try to port `name` table to use that.   But feel free to do this without that for now.  Thanks.